src/SFA.DAS.EAS.Portal.Client/PortalClient.cs

```diff
@@ -1,23 +1,63 @@
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using SFA.DAS.EAS.Portal.Client.Application.Queries;
+using SFA.DAS.EAS.Portal.Client.Services.DasRecruit;
 using SFA.DAS.EAS.Portal.Client.Types;
+using SFA.DAS.Encoding;
 using StructureMap;
 
 namespace SFA.DAS.EAS.Portal.Client
 {
     public class PortalClient : IPortalClient
     {
-        private readonly GetAccountQuery _getAccountQuery;
-
-        public PortalClient(IContainer container)
+        private readonly IEncodingService _encodingService;
+        private readonly IGetAccountQuery _getAccountQuery;
+        private readonly IDasRecruitService _dasRecruitService;
+        
+        public PortalClient(IContainer container, IEncodingService encodingService)
         {
-            _getAccountQuery = container.GetInstance<GetAccountQuery>();
+            _encodingService = encodingService;
+            _getAccountQuery = container.GetInstance<IGetAccountQuery>();
+            _dasRecruitService = container.GetInstance<IDasRecruitService>();
         }
         
-        public Task<Account> GetAccount(long accountId, CancellationToken cancellationToken = default)
+        public async Task<Account> GetAccount(string hashedAccountId,
+            AccountState accountState, CancellationToken cancellationToken = default)
         {
-            return _getAccountQuery.Get(accountId, cancellationToken);
+            var hasPayeScheme = (accountState & AccountState.HasPayeScheme) == AccountState.HasPayeScheme;
+
+            // we potentially map 1 more vacancy than necessary, but it keeps the code clean
+            var vacanciesTask = hasPayeScheme ? 
+                _dasRecruitService.GetVacancies(hashedAccountId, 2, cancellationToken) : null;
```

Comment:
Can the decorator pattern be used here to separate the concerns of Getting the vacancy data and merging with the account from the concerns of getting the account model.  Then the existing code does not need to be changed and the new code is independent and can be tested separately and removed/modified if the source of vacancy data changes.

Reply:
Agree although not a showstopper in my opinion. Could add a small tech debt task to cleanup

Reply:
I personally think that while doing what you suggest would separate concerns, it would also add unnecessary complexity. Remember that `PortalClient` is injected by the client, not on the portal side, so the decoration would have to be done within the website's DI container. If the website's DI library is changed, which could be out of our control, the new library might not support decoration.

We could add another layer within the client's code to inject a new account service and decorate that, but then we'd have to modify the existing code anyway.

Reply:
I thought that we provide a PortalClientRegistry so would the decorations not be setup in here. If we replaced StructureMap this would cease to work as is regardless wouldn't it ?

Reply:
We do, and yes we'd have to change the portal code to support a different DI library.

Reply:
all I am getting at is that there are further implications if we were to change the DI library and we possibly shouldn't let that stop us from using decorators

Reply:
A debate for another day 😄.